service/src/lib.rs

```diff
@@ -793,26 +793,75 @@ pub struct FullNodeHandles {
 	pub validation_service_handle: Option<consensus::ServiceHandle>,
 }
 
-/// Create a new Polkadot service for a light client.
-pub fn polkadot_new_light(config: Configuration) -> Result<
-	(TaskManager, Arc<RpcHandlers>), ServiceError
->
-{
-	new_light::<polkadot_runtime::RuntimeApi, PolkadotExecutor, _>(config)
+/// A builder for a node.
+pub struct NodeBuilder {
+	config: Configuration,
 }
 
-/// Create a new Kusama service for a light client.
-pub fn kusama_new_light(config: Configuration) -> Result<
-	(TaskManager, Arc<RpcHandlers>), ServiceError
->
-{
-	new_light::<kusama_runtime::RuntimeApi, KusamaExecutor, _>(config)
-}
+impl NodeBuilder {
+	/// Create a new node builder.
+	pub fn new(config: Configuration) -> Self {
+		Self {
+			config,
+		}
+	}
 
-/// Create a new Westend service for a light client.
-pub fn westend_new_light(config: Configuration, ) -> Result<
-	(TaskManager, Arc<RpcHandlers>), ServiceError
->
-{
-	new_light::<westend_runtime::RuntimeApi, KusamaExecutor, _>(config)
```

Comment:
Note that I have removed the light node builders. I wish I could do the same for the full builders but we still need them for the `start_collator`.